Move signals to redux store

kadamwhite opened the pull request:
Will update description later, but PR'ing for now to get eyes on it: this moves all signals into the redux store, and (vebosely) synchronizes the view to that store on each update.

deathbearbrown commented:
Functional tested and nothing is broken.  I am logging the store in console and when I update values on a primitive I see the changes in the store. 

Looking through code now. :)

deathbearbrown commented on src/js/model/primitives/marks/Mark.js:
```diff
@@ -66,8 +66,9 @@ Mark.prototype.init = function() {
-      update[k] = dl.extend(sg.init(propSg(this, k), p.value),
+    if (typeof p.value !== 'undefined') {
+      sg.init(propSg(this, k), p.value);
+      update[k] = dl.extend(sg.reference(propSg(this, k)),
```
this is so much to unpack here.  Can you put a comment around this block just to say that this is where the signals are initialized?

kadamwhite replied:
Yes, I can do that! Thanks, @deathbearbrown

deathbearbrown commented:
Nothing jumped out at me with the code at first glance.

kadamwhite commented:
@deathbearbrown cleaned up the code, added a couple unit tests for the action creators, and added that comment you requested. I believe this is ready for final review and merge when you have time.

deathbearbrown commented on src/js/components/mixins/Parse.jsx:
```diff
@@ -1,29 +1,8 @@
-        model.update();
-      }
-    });
+    model.parse();
```
Question... do we need this now? Maybe for a later refactor if we aren't doing anything other than `model.parse()` we should get rid of this?

kadamwhite replied:
We do need this for now, because a re-parse of the model is needed in order to have certain updates take effect. I absolutely support removing the parse mixin in favor of an action that triggers that re-parse down the road, it simply felt like it was out of scope for this PR.

deathbearbrown commented on src/js/model/primitives/marks/Mark.js:
```diff
@@ -99,7 +103,7 @@ Mark.prototype.dataset = function(id) {
   }
 
-  this.from = undefined;
+  delete this.from;
```
I don't think you need to delete from.  
It's already set as undefined in the Mark function.  I guess you could remove it from there and then add the property in here?

kadamwhite replied:
How I read this (and @arvind can confirm) is that this code is meant to unset `this.from` if the dataset method is called with arguments that cause it to fall through all the `else if`'s... so we either need to keep the `= undefined`, which feels janky b/c we try to avoid using undefined as a value, even tho it does work in strict mode; or delete the property, which feels more explicit and accomplishes the same end. Initializing it to undefined to begin with still feels weird, b/c just omitting it entirely ought to accomplish the same goals; but I fixed this lint error about use of `undefined` then decided I didn't want to get too deep in the weeds by addressing the other as well. I would revert this if you don't feel it's appropriate, but I don't think I'd do anything further to this method b/c this patch was just for lint cleanup and isn't related to the store or signals.
